eventmesh/ApiServerLockManager.js

```diff
@@ -0,0 +1,136 @@
+const Client = require('kubernetes-client').Client;
+const config = require('kubernetes-client').config;
+const _ = require('lodash');
+const Promise = require('bluebird');
+const eventmesh = require('./');
+const errors = require('../common/errors');
+const ETCDLockError = errors.ETCDLockError;
+const logger = require('../common/logger');
+const CONST = require('../common/constants');
+
+class ApiServerLockManager {
+  /*
+  This method checks whether lock is of write type.
+  returns true if lock is present and not expired and it's of type WRITE
+  */
+
+  isWriteLocked(resourceName) {
+    return eventmesh.server.getLockResourceOptions(CONST.RESOURCE_TYPES.LOCK, CONST.RESOURCE_NAMES.DEPLOYMENT_LOCKS, resourceName)
+      .then(options => {
+        const currentTime = new Date();
+        const lockDetails = JSON.parse(options);
+        const lockTime = new Date(lockDetails.lockTime);
+        const lockTTL = lockDetails.lockTTL ? lockDetails.lockTTL : Infinity;
+        if (lockDetails.lockType === CONST.ETCD.LOCK_TYPE.WRITE && ((currentTime - lockTime) < lockTTL)) {
```

Comment:
put a logger statement
`${resourceName} is locked by lockDetails.lockedResourceDetails.ResourceId for lockDetails.lockOperation`

Reply:
Done!